### PR TITLE
refactor(RHTAPREL-680): refactor task publish-index-image

### PR DESCRIPTION
**Pull request by theflockers**
This PR refactor the task publish-index-image to
use the `internal-request` script instead of
the `create-internal-request` task.

Signed-off-by: Leandro Mendes <lmendes@redhat.com>

### CODE REVIEW

File: pipelines/fbc-release/README.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ Tekton release pipeline to interact with FBC Pipeline
 | taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                    | Yes       | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision                 | The revision in the taskGitUrl repo to be used                                                           | Yes       | main                                                            |
 
+### Changes in 1.5.0
+- add a new task reference to the task `publish-index-image` and adjusts its task parameters
```
Comment (mmalina) [inline pipelines/fbc-release/README.md:21]: If I didn't know about this PR, I wouldn't understand that it's about switching from using `create-internal-request` task to a dedicated task that uses the internal-request script. Can you modify this to say that?

File: tasks/publish-index-image/README.md [hunk 2]
```diff
@@ -4,16 +4,19 @@ Publish a built FBC index image using skopeo
...
-| Name | Description | Optional | Default value |
-|------|-------------|----------|---------------|
-| targetIndex | targetIndex signing image | No | |
-| inputDataFile | Optional file to read data from | No | |
-| retries | Number of skopeo retries | Yes | "0" |
-| publishingCredentials | The credentials used to access the registries | Yes | "fbc-publishing-credentials" |
-| requestUpdateTimeout | Max seconds waiting for the status update | Yes | 360 |
+| Name           | Description                                                             | Optional | Default value |
+|----------------|-------------------------------------------------------------------------|----------|---------------|
+| dataPath       | Path to the JSON string of the merged data to use in the data workspace | Yes      | "data.json"   |
```
Comment (mmalina) [inline tasks/publish-index-image/README.md:9]: I would suggest removing the `""` here and also for `retries` - at least from what I've seen, we usually don't use those here and it's implied that everything is a string.

File: pipelines/fbc-release/fbc-release.yaml [hunk 3]
```diff
@@ -326,25 +326,16 @@ spec:
...
-              jsonKey: ".index_image_resolved"
-            - name: targetIndex
-              value: $(tasks.update-ocp-tag.results.updated-targetIndex)
-            - name: retries
-              value: "0"
+        - name: sourceIndex
+          value: $(tasks.extract-index-image.results.indexImageResolved)
+        - name: targetIndex
+          value: $(tasks.add-fbc-contribution-to-index-image.results.requestTargetIndex)
+        - name: retries
```
Comment (mmalina) [inline pipelines/fbc-release/fbc-release.yaml:337]: Two questions:

1. Why don't we want any retries? Wouldn't it be better to have something like `3`?
2. If we do want 0 retries, then why specify this when 0 is already the default for the task?

Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml:337]: I did not think about that as I was just moving the parameters around, but 3 sounds reasonable. I will change.

Comment (theflockers) [reply pipelines/fbc-release/fbc-release.yaml:337]: changed.

**Review by mmalina — APPROVED** [review]

**Review by johnbieren — APPROVED** [review]

**Pull request merged**